[html] Clean state between dialog positioning tests

dpogue opened the pull request:
Calling `reset()` at the end of every test to close the modal dialog will only be invoked if the test passes. If an expectation fails, the call to `reset()` will never run and the dialog will remain modally open on the page. This causes all further tests to fail because calling `showModal()` on the dialog while it is open is an error.

Instead, set up `reset()` to be called as a cleanup after every test regardless of whether the test passes or fails.

dpogue committed:
[html] Clean state between dialog positioning tests

Calling `reset()` at the end of every test to close the modal dialog
will only be invoked if the test passes. If an expectation fails, the
call to `reset()` will never run and the dialog will remain modally open
on the page. This causes all further tests to fail because calling
`showModal()` on the dialog while it is open is an error.

Instead, set up `reset()` to be called as a cleanup after every test
regardless of whether the test passes or fails.

domenic reviewed (approved):
Could be cleaner by just doing `this.add_cleanup(reset)`, but LGTM. Thanks for fixing!

dpogue commented:
Good point! I'll make that change :)

dpogue committed:
[html] Fix closing dialogs in showModal test

The `add_cleanup()` calls need to be registered outside of the
`assert_throws` block to ensure that the dialogs are actually closed.
